src/AstronomicalCalendar.ts

```diff
@@ -702,14 +702,16 @@ export class AstronomicalCalendar {
    *            The calendar to set.
    */
   public setDate(date: DateTime | Date | string | number): void {
+    const zone: string = this.geoLocation.getTimeZone();
+
     if (DateTime.isDateTime(date)) {
-      this.date = date;
+      this.date = date.setZone(zone);
```

Comment:
I wasn't sure about this at first. I first thought a user might want to pass a DateTime, with a different timezone intentionally. But now I believe that taking a new timezone via this function would be counter-intuitive in most cases.
Consider `cal.setDate(DateTime.fromISO('2020-04-26'))` which would cause timezone to be overridden.